### PR TITLE
Patch/inputtime

### PR DESCRIPTION
**Pull request by stevemdixon**
### Purpose
This PR updates `InputTime` to take a `disabled` flag.
### Things to Note
I'm using an `||` condition in the `disabled={}` attribute to check if its value is `disabled` or `submitting`. **This shouldn't introduce any breaking changes. 👍**
### Review
I'm keen to see how this one feels! 👍

### CODE REVIEW

File: InputTime/index.jsx [hunk 1]
```diff
@@ -115,20 +117,21 @@ const InputTime = ({
...
           <option key={min} value={min}>{leftPadTimeUnit(min)}</option>)}
       </select>
-      { select24Hours ? null : renderAmPm({ noStyle, onChange, submitting, value }) }
+      { select24Hours ? null : renderAmPm({ disabled, noStyle, onChange, submitting, value }) }
     </div>
   );
 };
 
 InputTime.propTypes = {
+  disabled: PropTypes.bool.isRequired,
```
Comment (hharnisc) [inline InputTime/index.jsx]: This will trigger a warning if this isn't set - most of the time this won't be so I don't think this should be required.

File: InputTime/index.jsx [hunk 2]
```diff
@@ -84,8 +86,8 @@ const InputTime = ({
       marginBottom: 0,
       marginLeft: 0,
       padding: 0,
-      '-webkit-appearance': 'none',
-      '-moz-appearance': 'none',
+      WebkitAppearance: 'none',
```
Comment (hharnisc) [inline InputTime/index.jsx]: Good to fix things like this when you're in the code - might be better to do this in another commit depending on what it is -- but this is probably small enough :smile: 

Curious if we still need noStyle here BTW -- last time we talked about the inputtime you where using the standard component

Comment (stevemdixon) [inline InputTime/index.jsx]: Great point! Yeah, I was thinking whether or not to update this in a new branch. I'll definitely isolate these small pieces in new branches or separate commits. Thanks for the nudge there! 😅

Yeah, here's a screenshot of how it's rendering in buffer-web:
screen shot 2017-06-01 at 9 34 13 pm

 We likely won't be able to drop `noStyle` until perhaps buffer-web goes full-width. I'm keen to explore other options here to keep `InputTime` clean!

File: InputTime/index.jsx (hunk 1 above)
Comment (stevemdixon) [inline InputTime/index.jsx]: Ah, yes. Totally overlooked that. Will remove the `isRequired` method here.

File: InputTime/index.jsx (hunk 2 above)
Comment (hharnisc) [inline InputTime/index.jsx]: I see what you're saying here, the default styling makes the select element wider :smile: 

As far as the `fontFamily` goes -- setting that explicitly on the `Text` component would likely fix that -- would be good to do that in a followup commit on buffer components :smile_cat:

Comment (stevemdixon) [inline InputTime/index.jsx]: Yeah! I'm curious what your hunch might be regarding styling `select` elements. I wonder if we could approach `InputTime` like we do with `Button` and have `InputTime` and then `<InputTime small />` -- both using the same UI except `small` being similar in size to what we have in the screenshot above. 🤔
